src/ext/java/org/opentripplanner/ext/siri/TimetableHelper.java

```diff
@@ -154,10 +154,7 @@ public static Result<TripTimes, UpdateError> createUpdatedTripTimes(
         }
 
         if (foundMatch) {
-          ZonedDateTime startOfService = ServiceDateUtils.asStartOfService(
-            serviceDate.toLocalDate(),
-            zoneId
-          );
+          ZonedDateTime startOfService = ServiceDateUtils.asStartOfService(serviceDate, zoneId);
```

Comment:
Can we move this and line 280 to line 128, as the `serviceDate` is now calculated outside the loop.

Reply:
Done